libsolidity/ast/Types.cpp

```diff
@@ -1869,12 +1830,35 @@ bool ArrayType::isDynamicallyEncoded() const
 	return isDynamicallySized() || baseType()->isDynamicallyEncoded();
 }
 
-bigint ArrayType::storageSizeUpperBound() const
+bigint ArrayType::sizeUpperBound(DataLocation _loc) const
 {
 	if (isDynamicallySized())
 		return 1;
-	else
+
+	if (_loc == DataLocation::Storage)
 		return length() * baseType()->storageSizeUpperBound();
+
+	if (_loc == DataLocation::CallData)
+		return unlimitedStaticCalldataSize(true);
+
+	assert(_loc == DataLocation::Memory);
+	bigint size = bigint(length());
+	auto type = m_baseType;
+	while (auto arrayType = dynamic_cast<ArrayType const*>(type))
+	{
+		if (arrayType->isDynamicallySized())
+			break;
+		else
+		{
+			size *= arrayType->length();
+			type = arrayType->baseType();
+		}
+	}
+	if (type->isDynamicallySized())
+		size *= type->memoryHeadSize();
```

Comment:
OK, let's keep it as is for the moment :).

Reply:
It's just 32 here, isn't it? (Referring to https://github.com/ethereum/solidity/pull/9746/files#r484533776 in case github messes up the comment placement again). The point is: it should always be fine here and never fail any assertions and always fit in ``unsigned``.